src/plugins_types.c

```diff
@@ -1755,6 +1702,11 @@ extern LY_ERR ly_type_store_empty(const struct ly_ctx *ctx, const struct lysc_ty
         const struct lysc_node *ctx_node, struct lyd_value *storage, struct lys_glob_unres *unres, struct ly_err_item **err);
 extern LY_ERR ly_type_compare_empty(const struct lyd_value *val1, const struct lyd_value *val2);
 
+/* plugins_types_enumeration.c */
+extern LY_ERR ly_type_store_enum(const struct ly_ctx *ctx, const struct lysc_type *type,
```

Comment:
I do not like all these externs much but I suppose it saves a bunch of headers (there could be one for all the internal types, actually) so whatever. But what would make more sense to me is define the whole plugin structure (an item in `ly_builtin_type_plugins`) in the plugin C file and make only that extern.

Reply:
yes, I agree, it will be necessary for external plugins, so I expect it will be unified then.